src/components/editor/Bindings/Store/create.ts

```diff
@@ -98,12 +101,33 @@ const getInitialState = (
     ...getInitialStateData(),
 
     initializeCollectionData: (currentCollection, persistedDraftId) => {
-        const { setCollectionData } = get();
+        const { setCollectionData, setSchemaInferenceDisabled } = get();
 
         if (currentCollection) {
             evaluateCollectionData(persistedDraftId, currentCollection).then(
-                (response) => setCollectionData(response),
-                () => setCollectionData(undefined)
+                (response) => {
+                    setCollectionData(response);
+
+                    if (response) {
+                        const writeSchemaKey = response.spec.hasOwnProperty(
+                            'writeSchema'
+                        )
```

Comment:
Thinking out loud
We are doing this check in two places. We should make a `schemaHlper` or something soon.